intel_extension_for_transformers/llm/runtime/graph/application/main_run.cpp

```diff
@@ -196,7 +210,17 @@ int main(int argc, char** argv) {
   if (params.model_arch == MODEL_LLAMA) {
     add_bos = true;
   }
-  auto embd_inp = ::model_tokenize(ctx, params.prompt, add_bos);
+  
+  std::vector<int> embd_inp;
+  if (params.model_arch == MODEL_CHATGLM2 || params.model_arch == MODEL_CHATGLM1) {
+      std::vector<std::string> prompts;
+      prompts.push_back(params.prompt);
+      std::string prompt = build_prompt(prompts);
+      embd_inp = ::model_tokenize(ctx, prompt, false);
+      embd_inp.insert(embd_inp.begin(), {64790, 64792}); // special prefix
```

Comment:
what are `64790` and `64792` for? `bos_id`?

Reply:
These are special prefix. The tokenizer of chatglm-2 must add them as special tag.